pkg/controllers/disruption/validation.go

```diff
@@ -80,37 +79,54 @@ func (v *Validation) IsValid(ctx context.Context, cmd Command) (bool, error) {
 		case <-v.clock.After(waitDuration):
 		}
 	}
-	// Get the current representation of the proposed candidates from before the validation timeout
-	// We do this so that we can re-validate that the candidates that were computed before we made the decision are the same
-	// We perform filtering here to ensure that none of the proposed candidates have blocking PDBs or do-not-evict/do-not-disrupt pods scheduled to them
-	validationCandidates, err := GetCandidates(ctx, v.cluster, v.kubeClient, v.recorder, v.clock, v.cloudProvider, v.ShouldDisrupt, v.queue)
+	validatedCandidates, err := v.validateCandidates(ctx, cmd.candidates...)
+	if len(validatedCandidates) == 0 || err != nil {
```

Comment:
Why would we want to continue if the number of candidates isn't 0, but isn't the set of candidates we started off with?

Reply:
`ValidateCandidates` returns an empty slice if the full set of candidates were not valid. If the candidates were all valid, the returned slice is the updated Candidate representations.

Reply:
I think @njtran's point is that a failed validation should be an error here. Reading this function, I would assume I either get: 1) No error and the mapped candidates or 2) No mapped candidates and an error. The in-between state is a little confusing. I wonder if a special "FailedValidation" error could handle this better. At a high-level, I wonder if we could replace `isValid` with a `FailedValidation` error generally.

Reply:
The other thing that I don't love but was trying to figure out a good solve for and couldn't think of anything: a `Validate` function returns back retrieved data. Realistically, you want to retrieve the data, map the data, then validate the data. I get this is a little difficult since we've coupled some of our validation in the GetCandidates() logic. More of a musing and to see if you can think of something better that has closer semantic meaning than just `ValidateCandidates`.

Reply:
We could definitely replace this with a `ValidationError`, basically just needed a mechanism to differentiate between a 'fatal' error that would be propogated back up the call stack and an error reported vs a non-fatal error where we just discard the consolidation attempt.